e2e/cypress/support/hooks.js

```diff
@@ -1,22 +0,0 @@
-// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
-// See LICENSE.txt for license information.
-
-// Updates the config with the provided values before the test suite and then resets the config to the previous value
-// after the test suite has finished.
-export function testWithConfig(config) {
-    before(() => {
-        let originalConfig;
-        cy.apiLogin('sysadmin');
-        cy.apiGetConfig().then((resp) => {
-            originalConfig = resp.body;
-        });
-
-        cy.apiUpdateConfig(config);
-
-        after(() => {
-            // # Logging in again since another user session may happen between before and after
-            cy.apiLogin('sysadmin');
-            cy.apiUpdateConfig(originalConfig);
-        });
-    });
-}
```

Comment:
In the past, we're avoiding cleaning up state on after/afterEach ([ref on Cypress best practices](https://docs.cypress.io/guides/references/best-practices.html#Using-after-or-afterEach-hooks)) but I find this approach helpful especially with the system config, and so I'm in favor of making use of this in several places as needed. How about let's follow the same approach for resetting user props?